sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -55,6 +56,21 @@ class ParquetFileFormat
   with DataSourceRegister
   with Logging
   with Serializable {
+  // Poor man's "static initializer". Scala doesn't have language support for static initializers,
+  // and it's important that we initialize `ParquetFileFormat.redirectParquetLogsViaSLF4J` before
+  // doing anything with the Parquet libraries. Rather than expect clients to initialize the
+  // `ParquetFileFormat` singleton object at the right time, we put that initialization in the
+  // constructor of this class. This method is idempotent, and essentially a no-op after its first
+  // call.
+  ParquetFileFormat.ensureParquetLogRedirection
+
+  // Java serialization will not call the default constructor. Make sure we call
+  // ParquetFileFormat.ensureParquetLogRedirection in deserialization by implementing this hook
+  // method.
+  private def readObject(in: ObjectInputStream): Unit = {
+    in.defaultReadObject
+    ParquetFileFormat.ensureParquetLogRedirection
```

Comment:
You could also call `ensureParquetLogRedirection` from some main class right? e.g. `class Executor`.